core/Array.carp

```diff
@@ -108,43 +110,46 @@ Returns `Nothing` if the array is empty.")
   (defn /= [a b]
     (not (= (the (Ref (Array a)) a) b)))
 
-  ; TODO unsafe!
-  (doc maximum "gets the maximum in an array (elements must support `<`).")
+  (doc maximum "gets the maximum in an array (elements must support `<` and `zero`)and wraps it in a `Just`.
+
+If the array is empty, returns `Nothing`")
   (defn maximum [xs]
```

Comment:
This implementation doesn’t quite work as expected. I suggest the following alternative implementation:

```clojure
  (doc maximum "gets the maximum in an array (elements must support `<`) and
wraps it in a `Just`.

If the array is empty, it returns `Nothing`.")
  (defn maximum [xs]
    (if (empty? xs)
      (Maybe.Nothing)
      (let-do [result (nth xs 0)
               n (length xs)]
        (for [i 1 n]
          (let [x (nth xs i)]
            (when (< result x)
              (set! result x))))
        (Maybe.Just @result))))
```

This would also keep the copying to a minimum!